File: LeetCode/Easy/1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {

        Set<Integer> key = new HashSet<>();
        Set<Integer> dup = new HashSet<>();
        int[] count = new int[2_001];

        for(int i = 0; i < arr.length; ++i) {
            count[arr[i] + 1000]++;
            key.add(arr[i] + 1000);
        }

        for(int k : key) {
            if(dup.contains(count[k])) return false;
            dup.add(count[k]);
        }

        return true;
    }
}

File: LeetCode/Easy/1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.kt
class Solution {
    fun uniqueOccurrences(arr: IntArray): Boolean {

        val count = IntArray(2_001)
        val checked = BooleanArray(2_001)
        val occurrenceSet = BooleanArray(1_001)

        for(i in 0 .. arr.size - 1) {
            count[arr[i] + 1000]++
        }

        for(i in 0 .. arr.size - 1) {
            val idx = arr[i] + 1000;

            if(checked[idx]) continue

            if(occurrenceSet[count[idx]]) return false
            occurrenceSet[count[idx]] = true
            checked[idx] = true
        }

        return true
        
    }
}